src/screens/NativeIntegration/NIInputService.tsx

```diff
@@ -0,0 +1,279 @@
+      const valueWithBigNumber = await Promise.all(updatedArr);
+
+      const list = await approvalList();
+      const approveTxList = list ? await Promise.all(list) : [];
+
+      const contractInterface = await integrationContract[fnName](...valueWithBigNumber);
+
+      setContractRes([
+        ...approveTxList,
+        { ...contractInterface, value: isPayable ? utils.parseUnits(amount.toString(), 18) : 0 },
+      ]);
+
+      const approveListWithValue = approveTxList.map((response) => {
+        return { ...response, value: 0 };
+      });
+
+      dispatch(
+        estimateTransactionsAction(
+          [...approveListWithValue, { ...contractInterface, value: isPayable ? amount?.toString() : 0 }],
+          chain,
+        ),
+      );
+      setIsSendTransaction(false);
+      logBreadcrumb('nativeIntegrationContractResponse', JSON.stringify(contractInterface));
+    } catch (e) {
+      setIsSendTransaction(false);
+      setContractRes(undefined);
+      Toast.show({
+        message: e,
+        emoji: 'warning',
+      });
+      logBreadcrumb('contractInput error!', e);
+    }
+  };
+
+  const approvalList = async () => {
+    const approvalFilter = sequence?.filter((res) => res?.isApprovalNeeded);
+
+    const listOfApproval = approvalFilter
+      ? await approvalFilter.map(async (approvalData, i) => {
+          const index = sequence.indexOf(approvalData);
+
+          const approvalContractInterface = etherspotService.getContract(chain, ERC20_CONTRACT_ABI, value[index]);
+          try {
+            const decimalVal = await getDecimals(approvalContractInterface);
+            return approvalContractInterface.encodeApprove(
+              contractData?.contract_address,
+              utils.parseUnits(value[approvalData?.approveAmountPosition].toString(), decimalVal),
+            );
+          } catch (e) {
+            Toast.show({
+              message: 'Please enter valid address',
+              emoji: 'warning',
+            });
+            return null;
+          }
+        })
+      : null;
+
+    return listOfApproval;
+  };
+
+  const getDecimalValue = async (bigNumberValue, index) => {
+    const position = sequence?.[index]?.decimalAddressPosition;
+    const approvalContractInterface =
+      position !== undefined ? etherspotService.getContract(chain, ERC20_CONTRACT_ABI, value[position]) : null;
+
+    const decimal = !approvalContractInterface ? 0 : await getDecimals(approvalContractInterface);
+
+    return utils.parseUnits(bigNumberValue.toString(), decimal);
+  };
+
+  const nativeIntegrationTransaction = async () => {
+    try {
+      setIsSendTransaction(true);
+      const res = await etherspotService.setTransactionsBatchAndSend(contractRes, chain);
+      setIsSendTransaction(false);
+      navigation.navigate(NI_TRANSACTION_COMPLETED, { transactionInfo: { chain: chain, ...res } });
+    } catch (error) {
+      setIsSendTransaction(false);
+      Toast.show({
+        message: error.toString(),
```

Comment:
This is likely going to return an error that everyone hates. Could we make this error message a bit more friendly please?

Reply:
Okay, I'll make error message